exporters/zipkin/src/main/java/io/opentelemetry/exporters/zipkin/ZipkinSpanExporter.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.exporters.zipkin;
+
+import static java.util.concurrent.TimeUnit.MICROSECONDS;
+import static java.util.concurrent.TimeUnit.NANOSECONDS;
+
+import io.opentelemetry.common.AttributeValue;
+import io.opentelemetry.common.AttributeValue.Type;
+import io.opentelemetry.sdk.trace.data.SpanData;
+import io.opentelemetry.sdk.trace.export.SpanExporter;
+import io.opentelemetry.trace.Span.Kind;
+import io.opentelemetry.trace.Status;
+import io.opentelemetry.trace.attributes.SemanticAttributes;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import zipkin2.Endpoint;
+import zipkin2.Span;
+import zipkin2.codec.BytesEncoder;
+import zipkin2.reporter.Sender;
+
+/**
+ * This class was based on the OpenCensus zipkin exporter code at
+ * https://github.com/census-instrumentation/opencensus-java/tree/c960b19889de5e4a7b25f90919d28b066590d4f0/exporters/trace/zipkin
+ */
+final class ZipkinSpanExporter implements SpanExporter {
+
+  private static final Logger logger = Logger.getLogger(ZipkinSpanExporter.class.getName());
+
+  // The naming follows Zipkin convention. For http see here:
+  // https://github.com/openzipkin/brave/blob/eee993f998ae57b08644cc357a6d478827428710/instrumentation/http/src/main/java/brave/http/HttpTags.java
+  // For discussion about GRPC errors/tags, see here:  https://github.com/openzipkin/brave/pull/999
+  // Note: these 3 fields are non-private for testing
+  static final String GRPC_STATUS_CODE = "grpc.status_code";
+  static final String GRPC_STATUS_DESCRIPTION = "grpc.status_description";
+  static final String STATUS_ERROR = "error";
+
+  private static final Endpoint EMPTY_ENDPOINT = Endpoint.newBuilder().build();
+
+  private final BytesEncoder<Span> encoder;
+  private final Sender sender;
+
+  ZipkinSpanExporter(BytesEncoder<Span> encoder, Sender sender) {
+    this.encoder = encoder;
+    this.sender = sender;
+  }
+
+  static Span generateSpan(SpanData spanData) {
+    Endpoint endpoint = EMPTY_ENDPOINT;
+    Map<String, AttributeValue> resourceAttributes = spanData.getResource().getAttributes();
```

Comment:
I understand where this is coming from and the discussions around it, but I still haven't heard any reason to believe this data would be set by default. I suspect that's why the jaeger export was written to have a gear for this. https://github.com/open-telemetry/opentelemetry-java/blob/c32c776a70007760aac9e410bae63ecf6f282081/exporters/jaeger/src/test/java/io/opentelemetry/exporters/jaeger/JaegerIntegrationTest.java#L85

Recall the main thing we've done recently is dodge a 2 arg factory method. To make data unqueryable is a more bitter pill than having everyone call the builder. Also, it is somewhat arbitrary to have a shortcut for a url, even if we did there are other ways to do that with AutoValue, for example, making a concrete builder method that forwards to the sender one. 

Ex we do similarly in our okhttp sender
```java
    /**
     * No default. The POST URL for zipkin's <a href="https://zipkin.io/zipkin-api/#/">v2 api</a>,
     * usually "http://zipkinhost:9411/api/v2/spans"
     */
    // customizable so that users can re-map /api/v2/spans ex for browser-originated traces
    public Builder endpoint(String endpoint) {
      if (endpoint == null) throw new NullPointerException("endpoint == null");
      HttpUrl parsed = HttpUrl.parse(endpoint);
      if (parsed == null) throw new IllegalArgumentException("invalid post url: " + endpoint);
      return endpoint(parsed);
    }
```

Because this is likely to result in unqueryable and unaggregatable data, and in the code at least in this repo I see nothing that would guarantee this service.name field is set. I strongly recommend doing the same as the jaeger one. Otherwise, it will create a very bad experience and not worth waiting for complaints about (which likely would dump into zipkin's channel)


a README update, mentioning what if anything sets this or how users can do that manually if their instrumentation does not.

Reply:
The idea, as I understand it, is that the person configuring the SDK will add the Resource to the TracerProvider once, at startup. It's not code that would exist in the SDK itself, but written by the SDK user. 

I'm not sure about that jaeger exporter, and why it might have been written that way. My guess is that it predates the semantic convention of putting service.name in the Resource.

I'll put an exporter-level serviceName back and have it be the fallback.

Reply:
There is an issue to use the resource for that in Jaeger

Reply:
cool maybe link to that in a TODO. regardless it seems a fallback is prudent.

Reply:
https://github.com/open-telemetry/opentelemetry-java/issues/978

Reply:
There are some concerns that you also pointed. We should resolve both exporters when we do it.

Reply:
Also we have this defined in specs as well: 
https://github.com/open-telemetry/opentelemetry-specification/pull/472

@adriancole may want to confirm that it is what you expect :). Thanks

Reply:
ok, I restored the serviceName option, but have it be superceded by the service.name set in the Resource, if it's there. Phew.

If this is an ok approach for now, then let's get this bad boy merged and see if it works!